phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -354,10 +361,14 @@ public boolean nextRaw(List<Cell> result) throws IOException {
                 overrideDelegate();
                 return super.nextRaw(result);
             }
+            @Override
+            public RegionScanner getNewRegionScanner(Scan scan) throws IOException {
+                return new RegionScannerHolder(c, scan,
+                        ((DelegateRegionScanner) delegate).getNewRegionScanner(scan));
```

Comment:
Coercing without checking the instance is of type DelegateRegionScanner can lead to ClassCastException.

Reply:
That is the expected behavior.

Reply:
Added try-catch for ClassCastException